### DISCUSSION TITLE
Does this also support Unreal Engine?

### DISCUSSION THREAD

**Discussion by jinhongee** [supabase-community/supabase-csharp#141] [Q&A]
Hi everyone,

I'm currently in the role of a server programmer for a unreal engine game development team. I recently ran into supabase and loved how simple it made the whole backend workflow. But I'm not sure if this supports unreal engine like it does so to unity.

If there's any advice, that would be greatly appreciated!

**Comment by wiverson** [comment]
I'm not as familiar with Unreal (I know Unity & Godot better, my background is Java since 1995), but here is my understanding:

Out of the box Unreal has Blueprints and C++. I did find this https://github.com/nxrighthere/UnrealCLR which you could try using with this library.

Worst case basically everything in Supabase is accessible either directly to Postgres (typical classic server side) and/or via REST services. So this C# library is basically a lot of wrappers to make things easier.

I mostly use the PostgREST stuff, not the client-side query builder stuff. So it's basically just the Auth wrappers and the endpoint to hit the REST endpoints. In practice I want to say I'm only hitting around 20 or so endpoints? So in theory you could build an Unreal plugin to wrap those endpoints, eg https://dev.epicgames.com/community/learning/tutorials/ZdXD/call-rest-api-using-http-json-from-ue5-c

As for a lot of this kind of stuff, if you wanted to build an open source plugin for Unity that would be great. Or you could look into hiring a consultant to help out.

**Reply by wiverson** [reply]
Here's the Blueprints version for making REST calls. https://medium.com/@Codeible/making-a-http-request-in-unreal-engine-blueprints-using-one-node-972b67f20347 looks a bit clunky but also pretty easy.
